masq/src/terminal_interface.rs

```diff
@@ -1,65 +1,59 @@
 // Copyright (c) 2019-2021, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
+#[cfg(not(test))]
+use crate::line_reader::IntegrationTestTerminal;
 use crate::line_reader::{TerminalEvent, TerminalReal};
-use linefeed::{Interface, ReadResult, Writer};
-use masq_lib::constants::MASQ_PROMPT;
-use masq_lib::intentionally_blank;
-use std::sync::Arc;
-
 #[cfg(test)]
 use linefeed::memory::MemoryTerminal;
-
-#[cfg(not(test))]
-use crate::line_reader::IntegrationTestTerminal;
-
 #[cfg(not(test))]
 use linefeed::DefaultTerminal;
+use linefeed::{Interface, ReadResult, Writer};
+use masq_lib::constants::MASQ_PROMPT;
+#[cfg(test)]
+use masq_lib::intentionally_blank;
+use std::sync::Arc;
 
 pub const MASQ_TEST_INTEGRATION_KEY: &str = "MASQ_TEST_INTEGRATION";
-pub const MASQ_TEST_INTEGRATION_VALUE: &str =
-    "3aad217a9b9fa6d41487aef22bf678b1aee3282d884eeb74b2eac7b8a3be8xzt";
+pub const MASQ_TEST_INTEGRATION_VALUE: &str = "3aad217a9b9fa6d41487aef22bf678b1aee3282d884eeb\
+74b2eac7b8a3be8xzt";
 
-//this is a layer with the broadest functionality, an object which is intended for you to usually work with at other
-//places in the code
+//This is the outermost layer which is intended for you to usually work with at other places in the codebase.
 
 pub struct TerminalWrapper {
     interface: Arc<Box<dyn MasqTerminal + Send + Sync>>,
 }
 
-impl TerminalWrapper {
-    pub fn lock(&self) -> Box<dyn WriterLock + '_> {
-        self.interface.provide_lock()
-    }
-
-    pub fn read_line(&self) -> TerminalEvent {
-        self.interface.read_line()
-    }
-
-    pub fn add_history_unique(&self, line: String) {
-        self.interface.add_history_unique(line)
+impl Clone for TerminalWrapper {
+    fn clone(&self) -> Self {
+        Self {
+            interface: Arc::clone(&self.interface),
+        }
     }
+}
 
+impl TerminalWrapper {
     pub fn new(interface: Box<dyn MasqTerminal + Send + Sync>) -> Self {
         Self {
             interface: Arc::new(interface),
         }
     }
 
-    #[cfg(test)]
-    pub fn configure_interface() -> Result<Self, String> {
-        Self::configure_interface_generic(Box::new(result_wrapper_for_in_memory_terminal))
+    pub fn lock(&self) -> Box<dyn WriterLock + '_> {
```

Comment:
Having different method names here (`lock` and `provide_lock`) is mildly confusing. Not a big deal, but a small one.